design/roadmap.md

```diff
@@ -1,179 +1,71 @@
+## Dependencies
 
-## Easily generate third-party registries
+* Users can configure Ksonnet to use parts on the local filesystem as dependencies.
 
-*Planned for 0.9.0* ([ #234](https://github.com/ksonnet/ksonnet/issues/234))
+## Default parts
 
-Users can create custom, third-party registries, but the `ks registry add` command will only recognize them if they adhere to the registry spec. In other words, they need to have the appropriate YAML files and directory structure.
+* Support for parts based on Kubernetes types will be included with Ksonnet. These parts will act as local dependencies and be upgraded through the registry.
 
-**A new command (e.g. `ks registry create`) which autogenerates registry scaffolding files, may make this easier and less error prone.**
+## Expose environment metadata to prototypes
 
-## Lazily download and update dependencies (`ks install`)
+* Users can reference the current namespace in components.
 
-*Planned for 0.9.0* ([#217](https://github.com/ksonnet/ksonnet/issues/217), [#237](https://github.com/ksonnet/ksonnet/issues/237))
+## JSON and YAML support
 
-Currently, `ks init` fails when the user doesn't have an internet connection, because it relies on two external dependencies:
-* The `k.libsonnet` for the `default` environment
-* The `incubator` registry
+* Users can add JSON, YAML, or Jsonnet file into `components/` and be able to apply them with `ks apply`. Only Jsonnet files can use parameters.
+* Users can use `ks` to convert JSON or YAML files to Jsonnet.
 
-To avoid this issue, **ksonnet needs to lazily initialize environments and registries**, rather than trying to set these up right away.
+## Component references
 
-**This plays into the idea of a `ks install` command, which will cut down on the number of version-controlled files for a given ksonnet app**. ksonnet dependencies will be `.gitignore`-ed. *Only their metadata* (e.g. versions) will be tracked.
+* Components can refer to parameters in other components.
 
-After a ksonnet app is downloaded (e.g. via `git clone`), `ks install` can be run to download and "flesh out" the necessary environments, registries, and packages. This is similar to `npm install`.
+## Apply
 
-**An associated command (e.g. `ks update`) should allow users to download the latest versions of a dependency.**
+* Users can apply their configurations in a consistent manner. Resources that do not need to be changed will untouched during the `ks apply` process.
 
-## Expose "environment metadata" to prototypes
+## Secrets
 
-*Planned for 0.9.0* ([#222](https://github.com/ksonnet/ksonnet/issues/222))
+* The Ksonnet team will supply guidance on managing secrets in Ksonnet. The team is currently investigating Bitnami's [sealed secrets](https://github.com/bitnami/sealed-secrets) as an approach.
 
-Nearly all ksonnet prototypes expect a `namespace` parameter. However, in the Jsonnet code, there is currently no way of referencing the namespace of a given ksonnet environment.
+## Diff
```

Comment:
Would this be any more clear? (Still remember your feedback on "improve output of ks diff")
```
## Refine output of `ks diff`
```